server/chat-plugins/smogtours.ts

```diff
@@ -0,0 +1,428 @@
+				return this.popupReply(`Invalid info URL: ${url}`);
+			}
+			let image;
+			if (rawImg) {
+				if (!Chat.linkRegex.test(rawImg)) {
+					return this.popupReply(`Invalid image URL: ${rawImg}`);
+				}
+				try {
+					const dimensions = await Chat.fitImage(rawImg, 200, 200);
+					image = {url: rawImg, dimensions: dimensions.slice(0, 1) as [number, number]};
+				} catch (e) {
+					return this.popupReply(`Invalid image URL: ${rawImg}`);
+				}
+			}
+			if (!rawShort.length || !rawDesc.length) {
+				return this.popupReply(`Must provide both a short description and a full description.`);
+			}
+			const tour: TourEvent = {
+				title, url, image, shortdesc: rawShort, desc: rawDesc, id: tourID || toID(title),
+			};
+			if (isEdit) {
+				const idx = section.tours.findIndex(t => t.id === tour.id);
+				if (idx < 0) {
+					return this.popupReply(`Tour not found. Create one first.`);
+				}
+				section.tours.splice(idx, 1);
+			}
+			section.tours.push(tour);
+			saveTours();
+			this.refreshPage(`tournaments-add`);
+		},
+		end(target, room, user) {
+			const [sectionID, tourID] = target.split(',').map(toID).filter(Boolean);
+			if (!sectionID || !tourID) {
+				return this.parse(`/help smogtours`);
+			}
+			const section = tours[sectionID];
+			if (!section) return this.popupReply(`Invalid section: "${sectionID}"`);
+			const idx = section.tours.findIndex(t => t.id === tourID);
+			const title = section.tours[idx].title;
+			if (idx < 0) {
+				return this.popupReply(`Tour with ID "${tourID}" not found.`);
+			}
+			section.tours.splice(idx, 1);
+			this.refreshPage(`tournament-view-${sectionID}-${tourID}`);
+			this.popupReply(`Tour "${title}" ended.`);
+		},
+		whitelist(target, room, user) {
+			this.checkCan('rangeban');
+			const [sectionID, targetID] = target.split(',').map(toID).filter(Boolean);
+			if (!sectionID || !targetID) {
+				return this.parse(`/help smogtours`);
+			}
+			const section = tours[sectionID];
+			if (!section) {
+				return this.errorReply(`Invalid section ID: "${sectionID}". Valid IDs: ${Object.keys(tours).join(', ')}`);
+			}
+			if (section.whitelist?.includes(targetID)) {
+				return this.errorReply(`That user is already whitelisted on that section.`);
+			}
+			if (!section.whitelist) section.whitelist = [];
+			section.whitelist.push(targetID);
+			this.privateGlobalModAction(
+				`${user.name} whitelisted ${targetID} to manage tours for the ${section.title} section`
+			);
+			this.globalModlog('TOUR WHITELIST', targetID);
+			saveTours();
+		},
+		unwhitelist(target, room, user) {
+			this.checkCan('rangeban');
+			const [sectionID, targetID] = target.split(',').map(toID).filter(Boolean);
+			if (!sectionID || !targetID) {
+				return this.parse(`/help smogtours`);
+			}
+			const section = tours[sectionID];
+			if (!section) {
+				return this.errorReply(`Invalid section ID: "${sectionID}". Valid IDs: ${Object.keys(tours).join(', ')}`);
+			}
+			const idx = section.whitelist?.indexOf(targetID);
```

Comment:
```suggestion
			const index = section.whitelist?.indexOf(targetID);
```
nitpick but i feel like reducing a 5 letter variable name to 3 isnt necessary when there isnt a higher level variable w/ the same name